feat: add implement openai client in middleware

ankittk opened the pull request:
Issue: https://github.com/calque-ai/go-calque/issues/8
- added OpenAI Integration: Added full ai.Client implementation with streaming, tool calling, multimodal support, and JSON schema validation
- added config options (temperature, max tokens, penalties, etc.) and API key management via environment variables
- documentation update, example and unit test added

nullboundary commented:
@ankittk Thanks for implementing this! I checked it out and reviewed it and discovered a few issues:

1. convertToOpenAITools() in ai/openai/openai.go manually reconstructs the tool parameters, however currently the tools schema (pkg/middleware/tools/format.go) is already in OpenAI format so its unnecessary. This function could be greatly simplified. Eventually we will probably move away from using the OpenAI tools format internally but for now we are, so no reason to do unnecessary conversions. 
2. when I ran the example I got this error: `failed to create chat completion stream: this model is not supported MaxTokens, please use MaxCompletionTokens` this might be just a GPT-5 problem, we might have to test to see if MaxCompletionTokens works for older models as well. 
3. when I fixed the MaxCompletionTokens in the applyChatConfig() I also got this error: `failed to create chat completion stream: this model has beta-limitations, temperature, top_p and n are fixed at 1, while presence_penalty and frequency_penalty are fixed at 0` which is definitely a GPT-5 issue.
4. There also seems to be an issue with all the clients where the if you use the WithConfig it overwrites the DefaultConfig completely rather then merging them. This is my mistake and I'll make an another issue for it.

ankittk commented:
@nullboundary i updated that, unfortunately my quota ends and i cant test anymore. 

I have updated the MaxCompletionTokens and also keep GPT5 as a separate entity,

can you review the code? I am new in learning AI so please don't mind the code

nullboundary commented:
Continuing this in pr #26
